cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgresSqlStructuredTable.java

```diff
@@ -106,6 +106,10 @@ public void upsert(Collection<Field<?>> fields) throws InvalidFieldException, IO
     throws InvalidFieldException, IOException {
     LOG.trace("Table {}: Read with multiple keys {}", tableSchema.getTableId(), multiKeys);
 
+    if (multiKeys.isEmpty()) {
```

Comment:
does a check like this exist for the HBase implementation?

Reply:
HBase / LevelDB implementation, which is the BufferingTable, already handled that.